### PR TITLE
NodeinfoService and useNodeinfo hook

### PR DESCRIPTION
**Pull request by srosset81**
Protocol specification: https://nodeinfo.diaspora.software

- New NodeinfoService (in a new `@semapps/nodeinfo` package)
- New `useNodeinfo` hook
- Documentation

Example of implementation: https://github.com/assemblee-virtuelle/activitypods/pull/110/commits/b28e033ba020539f5566f298955a68a64adc803e

### CODE REVIEW

File: src/frontend/packages/activitypub-components/src/hooks/useNodeinfo.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+import { fetchUtils } from 'react-admin';
+
+const useNodeinfo = (host, rel = 'http://nodeinfo.diaspora.software/ns/schema/2.1') => {
+  const [schema, setSchema] = useState();
+
+  useEffect(() => {
+    (async () => {
+      const protocol = host.includes(':') ? 'http' : 'https'; // If the host has a port, we are likely on HTTP
```
Comment (Laurin-W) [inline src/frontend/packages/activitypub-components/src/hooks/useNodeinfo.js:9]: why not use origin instead of host?

**Review by Laurin-W — COMMENTED** [review]
That looks cool! So with the `addLink` action, we could add additional schemas, if we are in need, right?

**Comment by srosset81** [comment]
> That looks cool! So with the `addLink` action, we could add additional schemas, if we are in need, right?

Yes, and we already use it to link to the application actor:
https://github.com/assemblee-virtuelle/activitypods/pull/110/commits/b28e033ba020539f5566f298955a68a64adc803e#diff-6664b2fb00e0cfbdfb55a169f3bdd192bebd546e01dc1fb71368f286c1a9189aR88-R92

This is part of a proposed extension for ActivityPub (FEP):
https://codeberg.org/fediverse/fep/src/branch/main/fep/2677/fep-2677.md

File: src/frontend/packages/activitypub-components/src/hooks/useNodeinfo.js (hunk 1 above)
Comment (srosset81) [reply src/frontend/packages/activitypub-components/src/hooks/useNodeinfo.js:9]: For `/.well-known` type URLs, the practice is to use the domain name.
An improvement could be to first check https, and then http if https is not available.

**Pull request merged**